lib/DDG/Meta/Data.pm

```diff
@@ -122,6 +125,26 @@ sub get_ia {
     return $m;
 }
 
+sub _satisfy {
+	my ($ia, $by, $lookup) = @_;
+	return $by->($ia, $lookup) if ref $by eq 'CODE';
+	my $pby = $ia->{$by};
+	ref $lookup eq 'CODE' ? $lookup->($pby) : $pby eq $lookup;
+}
+
+# get_ias(repo => 'goodies', dev_milestone => 'live'...)
+# Lookups combine as an AND operation.
+sub get_ias {
+	my (@lookups) = @_;
+	my @results = @ia_container;
+	foreach (pairs @lookups) {
```

Comment:
@zachthompson Maybe...

```perl

my $ias = filter_ias(
    perl_module => $some_func,
    perl_module => $another_func,
);
```

Where `$some_func` and `$another_func` are just predicate functions. You would almost always be able to write it as `attr => sub { $some_func($_[0]) && $another_func($_[0]) }` though.

Hmm, okay - so if any attributes are specified as ARRAYs then treat it as an OR? This won't handle the case of *different* ORed attributes though? E.g., `status => 'live', dev_milestone => 'live'` for either being live.